script/DeployAirdrop.s.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.25 <0.9.0;
+
+import { Airdrop, UserAmount } from "src/Airdrop.sol";
+
+import { ProxyAdmin } from "lib/openzeppelin-contracts/contracts/proxy/transparent/ProxyAdmin.sol";
+import { TransparentUpgradeableProxy } from
+    "lib/openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import { console } from "forge-std/console.sol";
+
+import { BaseScript } from "script/BaseScript.s.sol";
+import { BatchUpdate } from "script/BatchUpdate.sol";
+import { ProxyUtils } from "script/ProxyUtils.sol";
+
+// source .env && forge script script/DeployAirdrop.s.sol:DeployAirdrop -s "run(string)"
+// script/inputs/season-one-eigen-holesky.json --rpc-url $HOLESKY_RPC_URL --sender $DEPLOYER_ADDRESS --account
+// $DEPLOYER_ACCOUNT_NAME --broadcast --verify --etherscan-api-key $ETHERSCAN_API_KEY
+contract DeployAirdrop is BaseScript, BatchUpdate {
+    Airdrop public airdrop;
+    Airdrop public airdropImpl;
+    ProxyAdmin public proxyAdmin;
+    uint256 public constant BATCH_SIZE = 800;
+
+    error InvalidDeployment();
+
+    function run(string memory _path) public {
+        _loadInput(_path);
+
+        _deploy();
+        _verify();
+        _save();
+    }
+
+    function _deploy() internal {
+        vm.startBroadcast();
+
+        address deployer = msg.sender;
+        console.log("Deployer address: ", deployer);
+
+        airdropImpl = new Airdrop();
+
+        console.log("Deployed Airdrop implementation: ", address(airdropImpl));
+
+        TransparentUpgradeableProxy proxy =
+            new TransparentUpgradeableProxy(address(airdropImpl), data.proxyAdminOwner, "");
+
+        console.log("Deployed Airdrop proxy: ", address(proxy));
+
+        airdrop = Airdrop(address(proxy));
+
+        UserAmount[] memory _userAmounts = new UserAmount[](0);
+        airdrop.initialize(deployer, rewardsSafe, token, _userAmounts);
+
+        console.log("Initialized Airdrop");
+
+        airdrop.pause();
+        console.log("Paused Airdrop for updating user amounts");
```

Comment:
@danoctavian we need to pause the airdrop first, since `updateUserAmounts` will revert if not paused.